### PR TITLE
Detect CNI installation in local-up-cluster.sh

### LINKED ISSUE #108168: local-up-cluster.sh: doesn't install CNI plugins if needed
### What happened?

Since we switched from dockershim to containerd, my cluster seems to come up okay but pods get stuck in ContainerCreating. When I investigated further, I found

```
  Warning  FailedCreatePodSandBox  16s   kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = failed to setup network for sandbox "dcd18bc5497553ab129df05324c262920bdaff0f4d906719756d8d2f0e91a292": failed to find plugin "loopback" in path [/usr/lib/cni]
```

Most dependencies of launching a cluster are installed automatically. In this case, it seems we don't get the CNI plugins installed and set up as needed.

/cc @adisky 
/sig node network

### What did you expect to happen?

Pod should have been created as expected.

### How can we reproduce it (as minimally and precisely as possible)?

Try to create pods on a system with local-up-cluster without CNI plugins already installed.

### Anything else we need to know?

_No response_

### Kubernetes version

<details>

```console
ehashman@fedora:~/src/k8s$ cluster/kubectl.sh version
Client Version: version.Info{Major:"1", Minor:"24+", GitVersion:"v1.24.0-alpha.3.34+b68063fce7abf6", GitCommit:"b68063fce7abf62c69c257610d93ed3eb100956a", GitTreeState:"clean", BuildDate:"2022-02-16T19:03:24Z", GoVersion:"go1.17.1", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"24+", GitVersion:"v1.24.0-alpha.3.34+b68063fce7abf6", GitCommit:"b68063fce7abf62c69c257610d93ed3eb100956a", GitTreeState:"clean", BuildDate:"2022-02-16T19:03:24Z", GoVersion:"go1.17.1", Compiler:"gc", Platform:"linux/amd64"}
```

</details>

### Cloud provider

<details>
N/A
</details>

### OS version

<details>

```console
ehashman@fedora:~/src/k8s$ cat /etc/os-release 
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
ehashman@fedora:~/src/k8s$ uname -a
Linux fedora 5.10.0-11-amd64 #1 SMP Debian 5.10.92-1 (2022-01-18) x86_64 GNU/Linux
```

</details>

### Install tools

<details>
hack/local-up-cluster.sh
</details>

### Container runtime (CRI) and and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by adisky**
Signed-off-by: Aditi Sharma <adi.sky17@gmail.com>

#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
Detect CNI installation in hack/local-up-cluster.sh, if not it installs CNI plugin

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubernetes/issues/108168

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: hack/local-up-cluster.sh [hunk 1]
```diff
@@ -1096,6 +1096,14 @@ function install_cni {
 EOF
 }
 
+function detect_cni {
+  echo "Checking CNI Installtion at /opt/cni/bin"
```
Comment (sftim) [inline hack/local-up-cluster.sh:1100]:
```suggestion
  echo "Checking CNI Installation at /opt/cni/bin"
```

File: hack/local-up-cluster.sh [hunk 2]
```diff
@@ -1096,6 +1096,14 @@ function install_cni {
...
+  if [[ ! command -v /opt/cni/bin/loopback &> /dev/null ]]; then
+    echo "CNI Installtion not found at /opt/cni/bin"
```
Comment (sftim) [inline hack/local-up-cluster.sh:1102]:
```suggestion
    echo "CNI Installation not found at /opt/cni/bin"
```

File: hack/local-up-cluster.sh [hunk 3]
```diff
@@ -1096,6 +1096,14 @@ function install_cni {
 EOF
 }
 
+function detect_cni {
```
Comment (dims) [inline hack/local-up-cluster.sh:1099]: `detect_cni_if_needed`?

Comment (adisky) [reply hack/local-up-cluster.sh]: done

**Pull request merged**